src/Typist.jsx

```diff
@@ -56,14 +59,25 @@ export default class Typist extends Component {
   }
 
   shouldComponentUpdate(nextProps, nextState) {
+    if (this.props.direction !== nextProps.direction) {
+      return true;
+    }
     for (let idx = 0; idx < nextState.text.length; idx++) {
       const txt = this.state.text[idx];
       const ntxt = nextState.text[idx];
       if (txt !== ntxt && ntxt.length > 0) return true;
     }
     return this.state.isDone !== nextState.isDone;
   }
-
+  componentDidUpdate(prevProps) {
+    if (this.props.direction !== prevProps.direction) {
+      if (this.props.direction === 1) {
+        this.typeAllLines();
+      } else if (this.props.direction === -1) {
+        this.removeAllLines();
```

Comment:
i tried to implement the reset but didn't workout. so i added a check to not to change direction if its still in progress